pkg/apis/agent/v1/klusterletaddonconfig_types.go

```diff
@@ -16,84 +16,93 @@ import (
+	// GlobalProxy defines the cluster-wide proxy configuration of managed cluster.
+	// +optional
+	GlobalProxy ProxyConfig `json:"globalProxy,omitempty"`
+
+	SearchCollectorConfig      KlusterletAddonAgentConfigSpec `json:"searchCollector"`
+	PolicyController           KlusterletAddonAgentConfigSpec `json:"policyController"`
+	ApplicationManagerConfig   KlusterletAddonAgentConfigSpec `json:"applicationManager"`
+	CertPolicyControllerConfig KlusterletAddonAgentConfigSpec `json:"certPolicyController"`
+	IAMPolicyControllerConfig  KlusterletAddonAgentConfigSpec `json:"iamPolicyController"`
+
+	// ImageRegistry defined the custom registry address of the images.
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
+	// +optional
+	ImageRegistry string `json:"imageRegistry,omitempty"`
 
-	ImageRegistry    string `json:"imageRegistry,omitempty"`
-	ImageNamePostfix string `json:"imageNamePostfix,omitempty"`
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
 	// +kubebuilder:validation:MinLength=1
 	ImagePullSecret string `json:"imagePullSecret,omitempty"`
 
+	// DEPRECATED in release 2.4 and will be removed in release 2.5 since not used anymore.
 	// +kubebuilder:validation:Enum=Always;Never;IfNotPresent
 	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
-
-	// // ComponentTagMap contains the tag of each component
-	// ComponentTagMap map[string]string `json:"componentTagMap"`
-	// // ComponentImageMap contains the image name of each component
-	// ComponentImageMap map[string]string `json:"componentImageMap"`
-
-	// used for dev work only
-	ComponentOperatorImage string `json:"componentOperatorImage,omitempty"`
 }
 
-// KlusterletAddonConfigApplicationManagerSpec defines configuration for the ApplicationManager component
-type KlusterletAddonConfigApplicationManagerSpec struct {
-	Enabled bool `json:"enabled"`
-
+// ProxyConfig defines the global proxy env for OCP cluster
+type ProxyConfig struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests.  Empty means unset and will not result in an env var.
 	// +optional
-	ArgoCDCluster bool `json:"argocdCluster"`
-}
+	HTTPProxy string `json:"httpProxy,omitempty"`
 
-// KlusterletAddonConfigSearchCollectorSpec defines configuration for the SearchCollector component
-type KlusterletAddonConfigSearchCollectorSpec struct {
-	Enabled bool `json:"enabled"`
-}
+	// HTTPSProxy is the URL of the proxy for HTTPS requests.  Empty means unset and will not result in an env var.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
 
-// KlusterletAddonConfigCertPolicyControllerSpec defines configuration for the CertPolicyController component
-type KlusterletAddonConfigCertPolicyControllerSpec struct {
-	Enabled bool `json:"enabled"`
+	// NoProxy is a comma-separated list of hostnames and/or CIDRs for which the proxy should not be used.
+	// Empty means unset and will not result in an env var.
```

Comment:
It would  be  helpful if we doc commonly what  should be put in noproxy

Reply:
done